packages/nimble-components/src/select/index.ts

```diff
@@ -60,34 +198,174 @@ export class Select extends FoundationSelect implements ErrorPattern {
         this.updateListboxMaxHeightCssVariable();
     }
 
-    // Workaround for https://github.com/microsoft/fast/issues/5773
+    // Workaround for https://github.com/microsoft/fast/issues/5773d
+    // Additionally, we need to force an update to the filteredOptions observable
+    // (by calling 'filterOptions()) so that the template correctly updates.
     public override slottedOptionsChanged(
         prev: Element[],
         next: Element[]
     ): void {
         const value = this.value;
         super.slottedOptionsChanged(prev, next);
+        this.filterOptions();
         if (value) {
             this.value = value;
         }
     }
 
-    private regionChanged(
-        _prev: AnchoredRegion | undefined,
-        _next: AnchoredRegion | undefined
-    ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+    public override clickHandler(e: MouseEvent): boolean {
+        this.updateSelectedIndexFromFilteredSet();
+        super.clickHandler(e);
+        return true;
+    }
+
+    public inputClickHandler(e: MouseEvent): void {
+        e.stopPropagation(); // clicking in filter input shouldn't close dropdown
+    }
+
+    /**
+     * Handle content changes on the control input.
+     *
+     * @param e - the input event
+     * @internal
+     */
+    public inputHandler(e: InputEvent): boolean {
+        this.filter = this.filterInputElement?.value ?? '';
+        if (!this.committedSelectedOption) {
+            this.committedSelectedOption = this._options.find(
+                option => option.selected
+            );
+        }
+        this.filterOptions();
+        this.clearSelection();
+        if (this.committedSelectedOption) {
+            this.committedSelectedOption.selected = true;
+        }
```

Comment:
If the condition of the if-statement directly below (line 245) evaluates to `true`, then the assignment here is undone. Would it make sense to write the logic so that only one executes? Something like:

```ts
if (
    this.filteredOptions.length > 0
    && this.committedSelectedOption
    && !this.filteredOptions.includes(this.committedSelectedOption)
) {
    this.committedSelectedOption.selected = false;
    this.filteredOptions[0]!.selected = true;
} else if (this.committedSelectedOption) {
    this.committedSelectedOption.selected = true;
}
```